src/jbi/router.py

```diff
@@ -1,34 +1,84 @@
 """
 Router dedicated to Jira Bugzilla Integration APIs
 """
+import importlib
 import logging
-from typing import Optional
+import traceback
+from types import ModuleType
+from typing import Dict, List, Optional
 
-from fastapi import APIRouter, Depends, Request
-from fastapi.responses import HTMLResponse
+from fastapi import APIRouter, Body, Depends, Request
+from fastapi.responses import HTMLResponse, JSONResponse
 from fastapi.templating import Jinja2Templates
 
 from src.app import environment
 from src.jbi import configuration
+from src.jbi.bugzilla_objects import BugzillaBug, BugzillaWebhookRequest
+from src.jbi.model import ActionError, IgnoreInvalidRequestError
+from src.jbi.service import get_bugzilla
 
 templates = Jinja2Templates(directory="src/templates")
 
 api_router = APIRouter(tags=["JBI"])
 
-jbi_logger = logging.getLogger("src.jbi")
+jbi_logger = logging.getLogger("src.jbi.router")
 
 
-def execute_request(request, settings):
+def extract_current_action(  # pylint: disable=inconsistent-return-statements
+    bug_obj: BugzillaBug, action_map, settings
+):
+    """Find first matching action from bug's whiteboard list"""
+    potential_configuration_tags: List[
+        str
+    ] = bug_obj.get_potential_whiteboard_config_list()
+    if not potential_configuration_tags:
+        return {}
+    for tag in potential_configuration_tags:
+        value = action_map.get(tag.lower())
+        if value:
+            return value
+
+
+def execute_request(request: BugzillaWebhookRequest, action_map, settings):
     """Execute action"""
+    try:
+        if not request.bug:
+            raise IgnoreInvalidRequestError("no bug data received")
+
+        is_private_bug = request.bug.is_private
+        if is_private_bug:
+            raise IgnoreInvalidRequestError("private bugs are not valid")
+
+        bugzilla_client = get_bugzilla()
+        current_bug_info = bugzilla_client.getbug(request.bug.id)
+        bug_obj = BugzillaBug.parse_obj(current_bug_info.__dict__)
+        current_action = extract_current_action(bug_obj, action_map, settings)
+        if not current_action:
+            raise IgnoreInvalidRequestError("bug does not have matching config")
+
+        action_module: ModuleType = importlib.import_module(current_action["action"])
+        if not action_module:
+            raise IgnoreInvalidRequestError("action not found")
```

Comment:
I believe this was added as pre-commit/mypy wasn't happy about the potential of action_module being "Nonetype" when calling `init` in the following line. This `if` was added as an additional confirmation to avoid the "Nonetype" issue when calling `init`; however I can add an ignore for mypy instead and let it error otherwise--which would be unexpected at this point in the code anyway.

Reply:
I've updated this in the most recent commit.